pkg/reconciler/ingress/ingress.go

```diff
@@ -108,23 +109,24 @@ func (r *Reconciler) ObserveKind(ctx context.Context, ing *v1alpha1.Ingress) rec
 	return nil
 }
 
-func (r *Reconciler) FinalizeKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
-	return r.ObserveFinalizeKind(ctx, ing)
-}
-
-func (r *Reconciler) ObserveFinalizeKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
+func (r *Reconciler) ObserveDeletion(ctx context.Context, key types.NamespacedName) error {
 	logger := logging.FromContext(ctx)
-	logger.Infof("Deleting Ingress")
+	logger.Infof("Ingress deleted, updating config")
 
-	r.statusManager.CancelIngressProbing(ing)
+	r.statusManager.CancelIngressProbingByKey(key)
 
-	if err := r.caches.DeleteIngressInfo(ctx, ing.Name, ing.Namespace); err != nil {
+	if err := r.caches.DeleteIngressInfo(ctx, key.Name, key.Namespace); err != nil {
 		return err
 	}
 
 	return r.updateEnvoyConfig(ctx)
 }
 
+func (r *Reconciler) FinalizeKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
+	// Keeping this for now to keep finalizer based logic intact.
```

Comment:
I don't quite understand how keeping this method helps here, can you say a couple words more?

Reply:
We won't actively remove the finalizer, so if we drop this function I believe the object will get stuck in finalization forever.

Reply:
Aah, I see, what a pain - yeah makes sense